src/host_vm.rs

```diff
@@ -305,11 +306,15 @@ impl<T: GuestStagePagingMode> HostVmLoader<T> {
         };
         self.vm
             .add_measured_pages(current_gpa, kernel_pages.into_iter());
-        current_gpa = current_gpa.checked_add_pages(num_kernel_pages).unwrap();
+        current_gpa = current_gpa
+            .checked_add_pages_with_size(
+                (num_kernel_pages / 512) + ((num_kernel_pages % 512 != 0) as u64),
+                PageSize::Size2M,
+            )
```

Comment:
when we create the `HwMemRegion` for the kernel, we should just round up the size to 2M so that we don't have to do this. same for initramfs.

Reply:
Agreed, this will simplify the code.